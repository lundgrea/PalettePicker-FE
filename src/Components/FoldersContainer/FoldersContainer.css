.FoldersContainer{
  margin: auto;
  border-top: 2px solid black;
}

#foldersContainer-header-text {
  font-size: 35px;
  text-align: center;
  margin: 30px 0px 15px 0px;
}