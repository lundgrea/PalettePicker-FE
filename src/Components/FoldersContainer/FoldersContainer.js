import React from "react";
import Folder from "../Folder/Folder";
import "./FoldersContainer.css";

export const FoldersContainer = props => {
  const allFolders = props.folders.folders.map((folder, index) => {
    return (
      <Folder 
      key={index}
      id={folder.id} 
      name={folder.name} 
      palettes={folder.palettes} 
      getAFoldersPalettes={props.getAFoldersPalettes}
      deleteAFolder={props.deleteAFolder}/>
    );
  });
  return (
    <article className="FoldersContainer">
      <h2 id="foldersContainer-header-text">Saved Folders</h2>
        <div id="folders-scroll-section">
        {allFolders}
        </div>
    </article>
  );
};

export default FoldersContainer;
