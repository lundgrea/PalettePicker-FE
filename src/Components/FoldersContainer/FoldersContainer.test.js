import React from "react";
import { shallow } from "enzyme";
import { FoldersContainer } from "./FoldersContainer";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
let folders = {folders: [{name: "Cruising", id: 99}, {name: "Just Joshin", id: 33}]}

beforeEach(() => {
  wrapper = shallow(<FoldersContainer folders={folders} getAFoldersPalettes={jest.fn()} deleteAFolder={jest.fn()} />)
});


describe("FoldersContainer", () => {
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

