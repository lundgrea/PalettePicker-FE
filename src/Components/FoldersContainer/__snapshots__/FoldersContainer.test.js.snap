// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoldersContainer should match snapshot 1`] = `
<article
  className="FoldersContainer"
>
  <h2
    id="foldersContainer-header-text"
  >
    Saved Folders
  </h2>
  <div
    id="folders-scroll-section"
  >
    <Folder
      deleteAFolder={[MockFunction]}
      getAFoldersPalettes={[MockFunction]}
      id={99}
      key="0"
      name="Cruising"
    />
    <Folder
      deleteAFolder={[MockFunction]}
      getAFoldersPalettes={[MockFunction]}
      id={33}
      key="1"
      name="Just Joshin"
    />
  </div>
</article>
`;
